types/lua-json/index.d.ts

```diff
@@ -0,0 +1,26 @@
+// Type definitions for lua-json 1.0
+// Project: https://github.com/kcwiki/lua-json#readme
+// Definitions by: bitomic <https://github.com/bitomic>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/**
+ * Takes a JSON object and returns a Lua table as a string.
+ */
+export function format(
+    value: Record<string | number, unknown>,
+    options?: {
+        eol?: string;
+        singleQuote?: boolean;
+        spaces?: number;
+    },
+): string;
+
+/**
+ * Takes a Lua table (string) and returns a JSON object.
+ */
+export function parse(value: string): Record<string | number, unknown>;
```

Comment:
Could you make the return type of this function use a generic?

Not sure if this code works, but something like this:  
```suggestion
export function parse<T = Record<string | number, unknown>>(value: string): T;
```

Reply:
@tommy-josepovic I wasn't sure if I could take the freedom to do so since the package's source is written in javascript, and of course generics don't exist there. However, of course it would be better that way!

Just a question, if you don't mind me asking and sorry if it is quite basic haha, would it be better `<T = Record<string | number, unknown>>` or `<T extends Record<string | number, unknown>>`?

Also, thanks for your comment! Would it be okay if I add you to the authors list? 😄

Reply:
You're right, ``<T extends Record<string | number, unknown>>`` would provide default generic type and would also apply a restriction on the accepted generics, which is better than what I proposed.

Now that I think about it, I think ``<T extends object>`` is more appropriate. Using ``Record<string | number, unknown>>`` forces the interfaces used in the generic to use index signatures.

![image](https://user-images.githubusercontent.com/44372776/129571914-763c16d4-1903-4a1a-8b3f-772091c5cb1a.png)
![image](https://user-images.githubusercontent.com/44372776/129571996-6f0baac6-e11e-44c2-a6c1-7a26b454f877.png)

I think you could even replace all ``Record<string | number, unknown>>`` with the``object`` type.

Feel free to add me to the author's list!

Reply:
I may be wrong, but I think it is not advisable to use the `object` type? I couldn't find any specific reason for this besides [this SO answer](https://stackoverflow.com/a/65862513), so I may be wrong with that. But if it disallows types like `number` and `string`, then probably it is the best solution?

If you don't know anything about `object` being discouraged, or you think it is okay, then feel free to modify as needed! Added as an author. ^^

Reply:
I found [this discussion](https://github.com/typescript-eslint/typescript-eslint/issues/2068) about banning the use of ``object``. They mention it being bad practice to use it in projects, but being fine for APIs.

Reply:
My opinion was also based on eslint banning `object`, but couldn't find anything haha. Could you please go ahead and add both the generic and tests? Thank you! 😄

Reply:
it seems package stipulated its own intended types:
```md
type Json = null | boolean | number | string | Json[] | { [_: string]: Json }

format(
  value: Json,
  options?: {
    eol: string = '\n',
    singleQuote: boolean = true,
    spaces: null | number | string = 2,
  }
): string

parse(value: string): Json
```

Reply:
@peterblazejewicz I remember leaving a reply, weird. I forgot there was that on the README file, thank you!

Would you mind checking the tests file? I was unable to use `Json` type by any means, tried with `// $ExpectType Json`, `luajson.Json` and importing the type. Not sure what am I doing wrong, or if the only way is to define `Json` both in the declaration file and in the test file.

Thanks!